core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionExecutorIdImpl.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.util.compaction;
+
+import org.apache.accumulo.core.spi.compaction.CompactionExecutorId;
+import org.apache.accumulo.core.spi.compaction.CompactionServiceId;
+
+import com.google.common.base.Preconditions;
+
+public class CompactionExecutorIdImpl extends CompactionExecutorId {
```

Comment:
I assume this class was created to be separate from the SPI but I am not seeing anything internal exposed. It seems like it could just be included in the super class.

Reply:
I was trying to avoid putting all of the external/internal method in the SPI class.

Reply:
I get that. Thinking ahead is good but why make it more awkward right now if we don't have to? I was also thinking in this case having those "i" and "e" prefixes in the SPI would be good and there isn't much to `CompactionExecutorId`

Reply:
> why make it more awkward right now if we don't have to?

None of those methods are needed to write a compaction planner plugin, they are only used internally by the implementation outside of the realm of SPI interfaces.  So exposing them in the SPI could make changing the implementation harder in the future.

Reply:
To give some context using the DefaultPlanner as an example.  It will make SPI calls that give it a CompactionExecutorId like [here](https://github.com/apache/accumulo/blob/129d06b66ca52307a77f12cee203b70e3604585a/core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java#L126).  In never instantiates a CompactionExecutorId.  Then later it will pass those IDs back when planning like [here](https://github.com/apache/accumulo/blob/129d06b66ca52307a77f12cee203b70e3604585a/core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java#L240-L242)